client/js/table-view.js

```diff
@@ -47,43 +49,137 @@ class TableView{
 		const fragment = document.createDocumentFragment();
 		for(let row=0;row<this.model.numRows;row++){
 			const tr = createTR();
+			tr.appendChild(createTH(row+1));
 			for(let col=0;col<this.model.numCols;col++){
 				const position = {col:col,row:row};
 				const value = this.model.getValue(position);
 				const td = createTD(value);
-				if(this.isCurrentCell(col,row)){
+				if(this.isCurrentCell(col,row) && !this.checkLastRow(row)){
 					td.className = 'current-cell';
 				}
 
 				tr.appendChild(td);
 			}
 			fragment.appendChild(tr);
+			if (this.checkLastRow(row)){
+				tr.className = 'last-row';
+			}
 		}
 		removeChildren(this.sheetbodyEl);
 		this.sheetbodyEl.appendChild(fragment);
 	}
+	checkLastRow(row){
+		return row+1===this.model.numRows;
+	}
 	attachEventHandlers(){
 		this.sheetbodyEl.addEventListener('click',this.handleSheetClick.bind(this));
 		this.formulabarEl.addEventListener('keyup',this.handleFormulaBarChange.bind(this));
+		this.row.addEventListener('click',this.addRow.bind(this));
+		this.column.addEventListener('click',this.addColumn.bind(this));
+		this.headerRowEl.addEventListener('click',this.highlightColumn.bind(this));
+		//this.RowHeaderEl.addEventListener('click',this.highlightRow.bind(this));
 	}
 	isColumnHeaderRow(row){
 		return row < 1;
 	}
+	isRowHeaderColumn(col){
+		return col < 1;
+	}
+	addRow(){
+		this.model.numRows++;
+		this.renderTableBody();
+	}
+	addColumn(){
+		this.model.numCols++;
+		this.renderTable();
+	}
+	highlightColumn(e){
+		var colTarget = e.target.cellIndex;
+		var rowTarget = e.target.parentElement.rowIndex;
+		const fragment = document.createDocumentFragment();
+		for(let row=0;row<this.model.numRows;row++){
```

Comment:
Consistent spacing leads to easier readability for your human code readers. (Also, more spaces doesn't affect space or performance since you will minify your code before deploying is real production environments). Add spaces between operators and expressions.